### PR TITLE
Positioning precise view (aka stakeout) implementation

### PR DESCRIPTION
**Pull request by nirvn**
This PR implements a brand new precise view (aka stakeout) panel which appears when a navigation to destination is active. First, how it looks:
image

The precise view's precision (i.e. the target radius) can be customized by long tapping on the navigation button and selecting the precise view menu item:
image

Implementation details:
- the precise view will automatically appear when the positioning device's location falls within the precise view's precision value (set to 2.5 meter by default), however is is possible to have the precise view always show by toggling the [x] always show precise view menu item. the precise view essentially turns into a rotating compass with an arrow indicating the destination location
- a brand new proximity alarm is shipping with the precise view, which will play audible pings when the location falls within the precise view's max. precision value; the closer the distance, the faster the ping will play

The precise view target outline as well as the positioning location marker will turn green when having reached the target, which takes into consideration the device's accuracy _as well as_ the precise view's precision value (i.e., the higher the precision value (e.g. 0.25cm), the narrower the "reached destination" threshold will be:
image

When using a precise view's precision value that is too low for the positioning device accuracy,  a warning is shown:
image

When the navigation destination has a Z value (i.e. setting a PointZ feature as destination), the precise view will also display information about the vertical distance to destination (see the vertical bar at the right-end):
image

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for f2a4771fb22b16fba7818282496c3b11600b491d:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-a60897d80e191619e578bf196431f0aab88034be-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-a60897d80e191619e578bf196431f0aab88034be-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-a60897d80e191619e578bf196431f0aab88034be-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-a60897d80e191619e578bf196431f0aab88034be-x86-android.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 34399b71ce1ca07cc46df2dd29ec4febce01c5cc:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-2e110b547c12d6a45ff844cf569387a3ddfb0407-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-2e110b547c12d6a45ff844cf569387a3ddfb0407-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-2e110b547c12d6a45ff844cf569387a3ddfb0407-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-2e110b547c12d6a45ff844cf569387a3ddfb0407-x86-android.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 3685cbbf29679a35a4add9ce30be460f8b833e07:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-73273ffbc3dcd263e09d34219e7372678a6629d7-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-73273ffbc3dcd263e09d34219e7372678a6629d7-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-73273ffbc3dcd263e09d34219e7372678a6629d7-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-73273ffbc3dcd263e09d34219e7372678a6629d7-x86-android.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 0bb626dd126c8d772307226ff6e3f558098d0ea8:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-bf3684d34b9a393ac09e2682b04f070dc3f31fb0-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-bf3684d34b9a393ac09e2682b04f070dc3f31fb0-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-bf3684d34b9a393ac09e2682b04f070dc3f31fb0-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-bf3684d34b9a393ac09e2682b04f070dc3f31fb0-x86-android.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 3cc5d7f0312c72bd29d517b8b22170f4d35701af:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-1a666dbb1b4f22d0b6e6af8563877d5504ef6b8f-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-1a666dbb1b4f22d0b6e6af8563877d5504ef6b8f-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-1a666dbb1b4f22d0b6e6af8563877d5504ef6b8f-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-1a666dbb1b4f22d0b6e6af8563877d5504ef6b8f-x86-android.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 23e441a2b49010dccc95d3525a1d0274903b1a62:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-a18d08bd068072d7b0672d79383359dde007f431-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-a18d08bd068072d7b0672d79383359dde007f431-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-a18d08bd068072d7b0672d79383359dde007f431-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-a18d08bd068072d7b0672d79383359dde007f431-x86-android.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for e9e9b65892600d03b55d3d8991a96c5d5e03eb5f:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-e4f913c3d91cad6dcb896fe509507cd3ffc20510-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-e4f913c3d91cad6dcb896fe509507cd3ffc20510-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-e4f913c3d91cad6dcb896fe509507cd3ffc20510-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-e4f913c3d91cad6dcb896fe509507cd3ffc20510-x86-android.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 6a756f1476aeabf92c9d0ffd4e1ba395e86d765b:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-480bd46a589e5938125aa2ff52449b317fe1befb-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-480bd46a589e5938125aa2ff52449b317fe1befb-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-480bd46a589e5938125aa2ff52449b317fe1befb-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-480bd46a589e5938125aa2ff52449b317fe1befb-x86-android.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 02a0b6bba87533ed2500bae20e769a7a9db10620:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-0fb0c0e48478448b0c0dcdfb17325589a4459a13-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-0fb0c0e48478448b0c0dcdfb17325589a4459a13-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-0fb0c0e48478448b0c0dcdfb17325589a4459a13-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-0fb0c0e48478448b0c0dcdfb17325589a4459a13-x86-android.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for fa06dd3b3887f724ba120e8e11d40179aa3d4080:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-3855a174ff30052036f6e29c41ac0305ff0e7ea1-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-3855a174ff30052036f6e29c41ac0305ff0e7ea1-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-3855a174ff30052036f6e29c41ac0305ff0e7ea1-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-3855a174ff30052036f6e29c41ac0305ff0e7ea1-x86-android.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 6102a073350966e478af19fbf117be05ba95de62:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-85d5ff17edbcdf10eed1e762f4502b7e0edf547c-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-85d5ff17edbcdf10eed1e762f4502b7e0edf547c-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-85d5ff17edbcdf10eed1e762f4502b7e0edf547c-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-85d5ff17edbcdf10eed1e762f4502b7e0edf547c-x86-android.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for fa4a1bddb02b24fb8c53f253f82ddd9833abc90c:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-c581db1075e682c0b5a40183702e9713810a3288-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-c581db1075e682c0b5a40183702e9713810a3288-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-c581db1075e682c0b5a40183702e9713810a3288-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-c581db1075e682c0b5a40183702e9713810a3288-x86-android.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for ab17a3458cb842dfe0175384d361a3cb77d55c5f:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-4d1f6854a189bf7bdd7093f31da1f76527a30b78-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-4d1f6854a189bf7bdd7093f31da1f76527a30b78-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-4d1f6854a189bf7bdd7093f31da1f76527a30b78-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-4d1f6854a189bf7bdd7093f31da1f76527a30b78-x86-android.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for b2132f575c133e9f776d0071281a05e9cb3bebab:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-a49432ca8087ffafa3dcc0a1eb9ee932ff7b157a-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-a49432ca8087ffafa3dcc0a1eb9ee932ff7b157a-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-a49432ca8087ffafa3dcc0a1eb9ee932ff7b157a-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-a49432ca8087ffafa3dcc0a1eb9ee932ff7b157a-x86-android.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for ab864d64abdb05c36cc51c649f548258554d027c:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-6bc859b0cec483efb67fc0675e77e934a8d085a5-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-6bc859b0cec483efb67fc0675e77e934a8d085a5-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-6bc859b0cec483efb67fc0675e77e934a8d085a5-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-6bc859b0cec483efb67fc0675e77e934a8d085a5-x86-android.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for efc116f941d9e8b55dc7ba2cb9daf05a4c70a781:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-cdfd57dce46a845ab169c9c03d08f3936e6eb8e1-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-cdfd57dce46a845ab169c9c03d08f3936e6eb8e1-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-cdfd57dce46a845ab169c9c03d08f3936e6eb8e1-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-cdfd57dce46a845ab169c9c03d08f3936e6eb8e1-x86-android.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 5e472871cd374851573c12eaa9cc46ffdcfec54d:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-ae9724ff3685634ac2207e4ab02653d99a3ad4b5-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-ae9724ff3685634ac2207e4ab02653d99a3ad4b5-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-ae9724ff3685634ac2207e4ab02653d99a3ad4b5-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-ae9724ff3685634ac2207e4ab02653d99a3ad4b5-x86-android.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for efcfff273fd3be82a4f421566d3a20661ce851df:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-c02b6e89e094201c57e0544431decc5e6cf725e4-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-c02b6e89e094201c57e0544431decc5e6cf725e4-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-c02b6e89e094201c57e0544431decc5e6cf725e4-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-c02b6e89e094201c57e0544431decc5e6cf725e4-x86-android.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 18e10f66022e752051c080a57e877c4f5467e543:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-7272831dd0fc24283892af1dcac2bd9f42b57b58-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-7272831dd0fc24283892af1dcac2bd9f42b57b58-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-7272831dd0fc24283892af1dcac2bd9f42b57b58-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-7272831dd0fc24283892af1dcac2bd9f42b57b58-x86-android.apk)

**Comment by duke-nyuki** [comment]
Task linked: [QF-2028 Precision display for staking out](https://app.clickup.com/t/2192114/QF-2028)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 6d7744502435bf7766a155ab838007cb699cf1b5:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-9d2ff6ed9d2ae69acd7309c0c7f3585fb7718662-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-9d2ff6ed9d2ae69acd7309c0c7f3585fb7718662-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-9d2ff6ed9d2ae69acd7309c0c7f3585fb7718662-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-9d2ff6ed9d2ae69acd7309c0c7f3585fb7718662-x86-android.apk)

File: src/core/navigation.h [hunk 1]
```diff
@@ -93,9 +98,16 @@ class Navigation : public QObject
 
     QgsGeometry path() const { return mPath; }
     double distance() const { return mDistance; }
+    double verticalDistance() const { return mVerticalDistance; }
     QgsUnitTypes::DistanceUnit distanceUnits() const { return mDa.lengthUnits(); }
     double bearing() const { return mBearing; }
 
+    bool proximityAlarm() const { return mProximityAlarm; }
```
Comment (m-kuhn) [inline src/core/navigation.h:105]: Can you add docs here?

Comment (nirvn) [reply src/core/navigation.h:105]: Yep.

**Comment by nirvn** [comment]
@m-kuhn , added documentation for new (and pre-existing) Navigation class functions.

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 8e9baa0e4c9925db13cecd10f27ad67e77bf33c1:
  - [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-3899d73b3308974ba57ec931f3cba4e2b1dfe463-arm64-android.apk)

Other architectures: [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-3899d73b3308974ba57ec931f3cba4e2b1dfe463-arm-android.apk), [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-3899d73b3308974ba57ec931f3cba4e2b1dfe463-x64-android.apk), [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3003-3899d73b3308974ba57ec931f3cba4e2b1dfe463-x86-android.apk)

**Review by m-kuhn — APPROVED** [review]

**Comment by nirvn** [comment]
Alright, let's get our  people running nightly builds explore this new functionality!

We can do follow ups in individual, smaller PRs.

**Pull request merged**